OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -132,6 +137,16 @@ IEnumerable<CPos> GetAdjacentCells()
 
 		bool CanUnload()
 		{
+			if (checkTerrainType)
+			{
+				var tileSet = self.World.TileSet;
+				var tiles = self.World.Map.MapTiles.Value;
+				var terrainType = tileSet[tileSet.GetTerrainIndex(tiles[self.Location])].Type;
```

Comment:
Ok.  This still skips custom terrain types, which will make this work incorrectly on bridges and ore fields.

This will need to be replaced with `world.Map.GetTerrainInfo(self.Location).Type`